types/node.lua

```diff
@@ -90,105 +106,64 @@ function nodetypes.box:absolute ()
    return clone
 end
 
-function nodetypes.box:lineContribution ()
-   -- Regardless of the orientations, "width" is always in the
-   -- writingDirection, and "height" is always in the "pageDirection"
+--- Returns either the width or the height of the box.
+-- Regardless of the orientations, "width" is always in the writingDirection,
+-- and "height" is always in the "pageDirection"
+--
+-- @treturn SILE.types.length  The width or height of the box, depending on the orientation.
+function box:lineContribution ()
    return self.misfit and self.height or self.width
 end
 
-function nodetypes.box:outputYourself ()
+--- Output routine for a box.
+-- This is an abstract method that must be overridden by subclasses.
+--
+function box:outputYourself (_, _)
```

Comment:
Ah luacheck warning here. I should hould have used `box.outputYourself (_, _, _)`

Or really? Is there any weird benefit for this, rather than ignoring "212/self" in our .luacheckrc, so methods look always the same (with an implicit self, but they be called with it anyway).

Reply:
I objected to this a long while back but I think I've come around and would be fine with making that an automatic exception. It does make it easier when copying code from prototypes when some implementations may need self and others don't.